### PR TITLE
classification num_gpus types int => float & removed device check for "auto" num_gpus

### PR DESCRIPTION
**Pull request by BlueRexPY**

### CODE REVIEW

File: actableai/tasks/classification.py [hunk 1]
```diff
@@ -36,7 +36,7 @@ def run(
         biased_groups: List[str],
         debiased_features: List[str],
         residuals_hyperparameters: Optional[Dict],
-        num_gpus: int,
+        num_gpus: float,
```
Comment (cnpgs) [inline actableai/tasks/classification.py:39]: The number of GPUs should be an integer, and as such should be `int`

File: actableai/tasks/classification.py [hunk 2]
```diff
@@ -515,7 +515,7 @@ def run(
         cross_validation_max_concurrency: int = 1,
         residuals_hyperparameters: Optional[Dict] = None,
         drop_duplicates: bool = True,
-        num_gpus: int = 0,
+        num_gpus: float = 0,
```
Comment (cnpgs) [inline actableai/tasks/classification.py:518]: The number of GPUs should be an integer, and as such should be `int`

File: actableai/tasks/classification.py [hunk 3]
```diff
@@ -734,7 +734,7 @@ def run(
         df = handle_boolean_features(df)
 
         # Determine GPU type
-        device = "gpu" if num_gpus == "auto" or num_gpus > 0 else "cpu"
+        device = "gpu" if num_gpus > 0 else "cpu"
```
Comment (cnpgs) [inline actableai/tasks/classification.py:737]: Note that this essentially reverses https://github.com/Actable-AI/actableai-lib/pull/338

**Review by cnpgs — CHANGES_REQUESTED** [review]

File: actableai/tasks/classification.py (hunk 1 above)
Comment (BlueRexPY) [reply actableai/tasks/classification.py:39]: You are sure? Documentation (https://docs.ray.io/en/latest/ray-core/tasks/using-ray-with-gpus.html#fractional-gpus)

fractional numbers are needed for asynchronous tasks on the same GPU

the number in this case is taken from the config
https://github.com/Actable-AI/actableai-app/blob/master/docker/.env#LL174C1-L174C25


in the code you can also see how often ray using fractional numbers for num_gpus

https://github.com/ray-project/ray/blob/c4d62111d51780afd2a7ec1d68d8bf3dcf90e24d/doc/source/serve/scaling-and-resource-allocation.md?plain=1#L115
https://github.com/ray-project/ray/blob/c4d62111d51780afd2a7ec1d68d8bf3dcf90e24d/rllib/tuned_examples/dqn/atari-dist-dqn.yaml#LL33C17-L33C17
https://github.com/ray-project/ray/blob/c4d62111d51780afd2a7ec1d68d8bf3dcf90e24d/rllib/tuned_examples/dqn/atari-duel-ddqn.yaml#LL37C14-L37C14

File: actableai/tasks/classification.py (hunk 3 above)
Comment (BlueRexPY) [reply actableai/tasks/classification.py:737]: since now the data is taken from the config, "auto" cannot get into num_gpus

Comment (cnpgs) [reply actableai/tasks/classification.py:737]: Ah you also modified the backend, this makes more sense now

**Review by trunghlt — APPROVED** [review]

**Review by trunghlt — APPROVED** [review]

**Review by trunghlt — APPROVED** [review]

**Review by cnpgs — APPROVED** [review]

**Pull request merged**